app/_src/gateway/breaking-changes/38x.md

```diff
@@ -15,4 +15,53 @@ Review the [changelog](/gateway/changelog/#3800) for all the changes in this rel
 
 ## Breaking changes and deprecations
 
-TBA
\ No newline at end of file
+### New DNS client
+Starting 3.8.x, a new DNS client library was implemented and added to {{site.base_gateway}}. The new DNS client introduces and new standardized way to configure a service, and helps improve performance. It is disabled by default, but will be enabled by default in a future release. For more information, see [Migrate to the new DNS client](/gateway/migrate-to-new-dns-client/).
+
+### `kong.logrotate` configuration file no longer overwritten during upgrade
+The `kong.logrotate` configuration file will no longer be overwritten during upgrade.
+When upgrading, set the environment variable `DEBIAN_FRONTEND=noninteractive` on Debian/Ubuntu to avoid any interactive prompts and enable fully automatic upgrades.
+
+### `log_statistics` defaults to `false` for AI Proxy
+A configuration validation was added to the [AI Proxy plugin](/hub/kong-inc/ai-proxy/) to prevent users from enabling `log_statistics` for
+providers that don't support statistics. In addition, the default of `log_statistics` was changed from
+`true` to `false`, and a database migration is added as well for disabling `log_statistics` if it
+has already been enabled upon unsupported providers.
+
+### Custom plugins that used shared Redis config
+
+In 3.8.0.0, Kong has changed and refactored the shared Redis configuration that previously was imported by `require "kong.enterprise_edition.redis"`. If you created a custom plugin that is using this shared configuration or if you have a forked versionof a plugin, like `rate-limiting-advanced`, then you might need to do additional steps before you can upgrade to the new version of this Redis config.
+
+Out-of-the-box the custom plugins should still work since the old shared configuration was left in place. The new config adds the `cluster_max_redirections` option for Redis Cluster, and the `cluster_nodes` format and
```

Comment:
```suggestion
Out of the box, custom plugins should still work since the old shared configuration is still available. The new config adds the `cluster_max_redirections` option for Redis Cluster, and the `cluster_nodes` format and
```